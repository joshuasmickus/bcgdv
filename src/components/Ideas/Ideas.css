.ideas {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100%;
  margin: 0;
  max-width: 100%;
  padding: 0;
  width: 100%;
}
