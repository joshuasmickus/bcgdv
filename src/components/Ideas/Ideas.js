import React, { PureComponent } from 'react';
import { arrayOf, shape } from 'prop-types';

import { ideaShape } from '../../Constants/PropTypes';

import Idea from '../Idea';
import IdeaForm from '../IdeaForm';

export class Ideas extends PureComponent {
  render() {
    const { ideas } = this.props;

    return (
      <ul className="ideas">
        {ideas.map(idea => {
          if (idea.isEditing) {
            return <IdeaForm key={idea.id} idea={idea} />
          }

          return <Idea key={idea.id} {...idea} />;
        })}
      </ul>
    );
  }
}

Ideas.propTypes = {
  ideas: arrayOf(shape(ideaShape))
};
