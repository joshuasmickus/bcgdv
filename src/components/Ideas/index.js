import './Ideas.css';

export { Ideas as default } from './Ideas';
