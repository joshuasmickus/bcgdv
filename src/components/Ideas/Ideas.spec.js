import React from 'react';
import { shallow } from 'enzyme';

import { Ideas } from './Ideas';

const ideas = [{
  id: '11',
  'created_date': 'some date',
  title: 'title here',
  body: 'body here'
},
{
  id: '12',
  'created_date': 'some date',
  title: 'title here',
  body: 'body here'
},
{
  id: '13',
  'created_date': 'some date',
  title: 'title here',
  body: 'body here',
  isEditing: true
}];

describe('Given an Ideas component', () => {
  describe('when it is rendered', () => {
    it('should match the snapshot', () => {
      const wrapper = shallow(<Ideas ideas={ideas} />);

      expect(wrapper.getElement()).toMatchSnapshot();
    });
  });
});
