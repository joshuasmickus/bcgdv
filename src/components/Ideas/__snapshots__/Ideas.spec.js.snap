// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given an Ideas component when it is rendered should match the snapshot 1`] = `
<ul
  className="ideas"
>
  <Connect(Idea)
    body="body here"
    created_date="some date"
    id="11"
    title="title here"
  />
  <Connect(Idea)
    body="body here"
    created_date="some date"
    id="12"
    title="title here"
  />
  <Connect(IdeaForm)
    idea={
      Object {
        "body": "body here",
        "created_date": "some date",
        "id": "13",
        "isEditing": true,
        "title": "title here",
      }
    }
  />
</ul>
`;
